src/charm.py

```diff
@@ -249,10 +254,24 @@ def _db_connection_string(self) -> str:
 
     def _set_proxy(self):
         """Set Squid proxy environment variables if configured."""
-        proxy_url = os.environ.get("JUJU_CHARM_HTTP_PROXY")
-        if proxy_url:
-            os.environ["HTTP_PROXY"] = proxy_url
-            os.environ["HTTPS_PROXY"] = proxy_url
+        http_proxy_env = os.environ.get("JUJU_CHARM_HTTP_PROXY")
```

Comment:
`get` supports a default value if the key is missing so you could combine these lines with the ones on 263, 264 as:
```python
http_proxy_str = os.environ.get("JUJU_CHARM_HTTP_PROXY", "")
```